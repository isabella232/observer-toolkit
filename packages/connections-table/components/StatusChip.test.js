import React from 'react'
import { renderWithTheme } from '@libp2p-observer/testing'
import StatusChip from './StatusChip'

describe('status chip', () => {
  it('throws if trying to render an invalid status', () => {
    expect(() =>
      renderWithTheme(<StatusChip status="invalid status" />)
    ).toThrow(
      new Error(
        'Chip option "invalid status" not in "ACTIVE", "OPENING", "CLOSING", "CLOSED", "ERROR"'
      )
    )
  })

  it('renders ACTIVE status as expected', () => {
    const { asFragment } = renderWithTheme(<StatusChip status="ACTIVE" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
