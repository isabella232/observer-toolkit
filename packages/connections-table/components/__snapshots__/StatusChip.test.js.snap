// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`status chip renders ACTIVE status as expected 1`] = `
.c1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  vertical-align: middle;
  width: 20px;
  height: 20px;
  border-radius: 20px;
}

.c0 {
  display: inline-block;
  padding: 0 4px;
  background: rgba(0,99,109,0.15);
  color: rgba(0,99,109,1);
  white-space: nowrap;
  font-size: 8pt;
  font-weight: 600;
}

.c2 {
  line-height: 1em;
  display: inline-block;
  padding: 8px 4px;
  vertical-align: middle;
}

<span
  className="c0"
>
  <span
    className="c1"
    size={20}
  >
    <svg
      viewBox="0 0 96 96"
    >
      <g
        data-name="Layer 34"
      >
        <path
          d="M48 19a29 29 0 11-29 29 29 29 0 0129-29m0-4a33 33 0 1033 33 33 33 0 00-33-33z"
          fill="currentColor"
        />
        <path
          d="M35.32 46.81l8.73 11.46 16.63-20.54"
          fill="none"
          stroke="currentColor"
          strokeMiterlimit={10}
          strokeWidth={4}
        />
      </g>
    </svg>
  </span>
  <span
    className="c2"
  >
    ACTIVE
  </span>
</span>
`;
